core-httpclient-impl/src/main/java/com/optimizely/ab/odp/DefaultODPApiManager.java

```diff
@@ -0,0 +1,179 @@
+        StringBuilder segmentsString = new StringBuilder();
+        for (int i = 0; i < segmentsList.size(); i++) {
+            if (i > 0) {
+                segmentsString.append(", ");
+            }
+            segmentsString.append("\\\"").append(segmentsList.get(i)).append("\\\"");
+        }
+        return segmentsString.toString();
+    }
+
+    // ODP GraphQL API
+    // - https://api.zaius.com/v3/graphql
+    // - test ODP public API key = "W4WzcEs-ABgXorzY7h1LCQ"
+    /*
+
+     [GraphQL Request]
+
+     // fetch info with fs_user_id for ["has_email", "has_email_opted_in", "push_on_sale"] segments
+     curl -i -H 'Content-Type: application/json' -H 'x-api-key: W4WzcEs-ABgXorzY7h1LCQ' -X POST -d '{"query":"query {customer(fs_user_id: \"tester-101\") {audiences(subset:[\"has_email\",\"has_email_opted_in\",\"push_on_sale\"]) {edges {node {name state}}}}}"}' https://api.zaius.com/v3/graphql
+     // fetch info with vuid for ["has_email", "has_email_opted_in", "push_on_sale"] segments
+     curl -i -H 'Content-Type: application/json' -H 'x-api-key: W4WzcEs-ABgXorzY7h1LCQ' -X POST -d '{"query":"query {customer(vuid: \"d66a9d81923d4d2f99d8f64338976322\") {audiences(subset:[\"has_email\",\"has_email_opted_in\",\"push_on_sale\"]) {edges {node {name state}}}}}"}' https://api.zaius.com/v3/graphql
+     query MyQuery {
+       customer(vuid: "d66a9d81923d4d2f99d8f64338976322") {
+         audiences(subset:["has_email","has_email_opted_in","push_on_sale"]) {
+           edges {
+             node {
+               name
+               state
+             }
+           }
+         }
+       }
+     }
+     [GraphQL Response]
+
+     {
+       "data": {
+         "customer": {
+           "audiences": {
+             "edges": [
+               {
+                 "node": {
+                   "name": "has_email",
+                   "state": "qualified",
+                 }
+               },
+               {
+                 "node": {
+                   "name": "has_email_opted_in",
+                   "state": "qualified",
+                 }
+               },
+                ...
+             ]
+           }
+         }
+       }
+     }
+
+     [GraphQL Error Response]
+     {
+       "errors": [
+         {
+           "message": "Exception while fetching data (/customer) : java.lang.RuntimeException: could not resolve _fs_user_id = asdsdaddddd",
+           "locations": [
+             {
+               "line": 2,
+               "column": 3
+             }
+           ],
+           "path": [
+             "customer"
+           ],
+           "extensions": {
+             "classification": "InvalidIdentifierException"
+           }
+         }
+       ],
+       "data": {
+         "customer": null
+       }
+     }
+    */
+    @Override
+    public String fetchQualifiedSegments(String apiKey, String apiEndpoint, String userKey, String userValue, List<String> segmentsToCheck) {
+        HttpPost request = new HttpPost(apiEndpoint);
```

Comment:
I see this as the lowest level method that will make the actual call that is why it's getting the full endPoint. Users might provide their own implementation so it made more sense for it to expect the full endPoint. My plan is to build this endpoint using the host and path in the `ODPSegmentManager` and pass it to this one. What do you think?

Reply:
That's what I thought. If we plan to support custom endpoints, this makes sense. Let's move on and review again at the top level. 
One thing we also should consider is that if clients want to implement their own segment services (replacing all our default ODPManager), they can determine segments using their own solution and directly feed the "qualifedSegments" into OptimizelyUserContext, instead of injecting own ODPManager into our SDKs.